lib/sitemap/sitemap.rb

```diff
@@ -8,18 +8,23 @@ def initialize(directory, timestamp = Time.now.utc)
     @timestamp = timestamp
   end
 
+  def generate_and_replace(search_config)
+    replace(generate(search_config))
+  end
+
   def generate(search_config)
     FileUtils.mkdir_p(@output_path)
 
-    # generate and link the sitemap data files
     sitemap_writer = SitemapWriter.new(@output_path, @timestamp)
-    sitemap_filenames_with_linkname = sitemap_writer.write_sitemaps(search_config)
-    update_links(sitemap_filenames_with_linkname)
+    sitemaps = sitemap_writer.write_sitemaps(search_config)
+    index = write_index(sitemaps.map(&:last))
```

Comment:
Is `index` a filename, or the name of an index?

Reply:
It's the filename of the sitemap index (https://www.gov.uk/sitemap.xml)

Reply:
Ah I see, in the context of search api index always makes me think of search indexes.